lms/djangoapps/courseware/views.py

```diff
@@ -1230,3 +1236,104 @@ def course_survey(request, course_id):
         redirect_url=redirect_url,
         is_required=course.course_survey_required,
     )
+
+
+@require_POST
+def generate_user_cert(request, course_id):
+    """
+    It will add the add-cert request into the xqueue.
+
+     Arguments:
+        request (django request object):  the HTTP request object that triggered this view function
+        course_id (unicode):  id associated with the course
+
+    Returns:
+        returns json response
+    """
+
+    if not request.user.is_authenticated():
+        log.info(u"Anon user trying to generate certificate for %s", course_id)
+        return JsonResponseBadRequest(_('You must be logged-in to generate certificate'))
+
+    student = request.user
+
+    # checking course id
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
```

Comment:
`SlashSeparatedCourseKey` is deprecated; please use `CourseKey.from_string` instead.  You'll also need to catch any `InvalidKey` exceptions that get raised here.